src/main/java/seedu/address/logic/commands/medical/EditMedicalCommand.java

```diff
@@ -0,0 +1,313 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_IMMUNIZATION_HISTORY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MEDICATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SURGERIES;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WEIGHT;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.CommandType;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.medical.Age;
+import seedu.address.model.medical.BloodType;
+import seedu.address.model.medical.Ethnicity;
+import seedu.address.model.medical.FamilyHistory;
+import seedu.address.model.medical.Gender;
+import seedu.address.model.medical.Height;
+import seedu.address.model.medical.Illnesses;
+import seedu.address.model.medical.ImmunizationHistory;
+import seedu.address.model.medical.Medical;
+import seedu.address.model.medical.Medication;
+import seedu.address.model.medical.Surgeries;
+import seedu.address.model.medical.Weight;
+import seedu.address.model.patient.Nric;
+
+/**
+ * Edits the details of an existing medical information in MedBook.
+ */
+public class EditMedicalCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+    public static final CommandType COMMAND_TYPE = CommandType.MEDICAL;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the details of the medical information identified "
+            + "by the index number used in the displayed medical information list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NRIC + "NRIC "
+            + PREFIX_AGE + "AGE "
+            + PREFIX_BLOODTYPE + "BLOODTYPE "
+            + PREFIX_MEDICATION + "MEDICATION "
+            + PREFIX_HEIGHT + "HEIGHT "
+            + PREFIX_WEIGHT + "WEIGHT "
+            + PREFIX_ILLNESSES + "ILLNESSES "
+            + PREFIX_SURGERIES + "SURGERIES "
+            + PREFIX_FAMILY_HISTORY + "FAMILY_HISTORY "
+            + PREFIX_IMMUNIZATION_HISTORY + "IMMUNIZATION_HISTORY "
+            + PREFIX_GENDER + "GENDER "
+            + PREFIX_ETHNICITY + "ETHNICITY\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NRIC + "S1234567L "
+            + PREFIX_BLOODTYPE + "B";
+
+    public static final String MESSAGE_EDIT_MEDICAL_SUCCESS = "Edited Medical Information: %1$s";
+
+    private final Index targetIndex;
+    private final EditMedicalCommand.EditMedicalDescriptor editMedicalDescriptor;
```

Comment:
I believe the EditMedicalCommand. is redundant?
```suggestion
    private final EditMedicalDescriptor editMedicalDescriptor;
```

Reply:
Removed